prusti-specs/src/export_spec_rewriter.rs

```diff
@@ -0,0 +1,336 @@
+use proc_macro2::{TokenStream, TokenTree};
+use quote::{quote, quote_spanned, ToTokens};
+use syn::spanned::Spanned;
+
+// generate a hash of path
+fn hash_path(path: syn::Path) -> u64 {
+    use std::collections::hash_map::DefaultHasher;
+    use std::hash::{Hash, Hasher};
+    println!("path to hash: {:?} \n", &path);
+    let mut hasher = DefaultHasher::new();
+    path.hash(&mut hasher);
+    hasher.finish()
+}
+
+// generate a macro identifier with given path in this format:
+// {last_segment in path}_{hash(path)}
```

Comment:
Why only the last one? Is this to make the identifier still semi-readable but also not too long?

Reply:
Yes, that's what I intend to do, putting up the full path makes it too long and it's not that useful (the info is in hash anyway)